### PR TITLE
:book: add notes about cert-managers CRD storage version deprecation

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

Adds information about the issue described in https://github.com/kubernetes-sigs/cluster-api/issues/6674#issuecomment-1162746050 to the documentation and adds a link to the documentation of Cert Manager.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #6674

### CODE REVIEW

**Comment by chrischdi** [comment]
@sbueringer / @fabriziopandini cherry pick for release-v1.2?

File: docs/book/src/developer/providers/v1.1-to-v1.2.md [hunk 1]
```diff
@@ -26,6 +26,15 @@ in ClusterAPI are kept in sync with the versions used by `sigs.k8s.io/controller
 - github.com/onsi/gomega: v0.17.0 => v0.18.1 (derived from controller-runtime)
 - k8s.io/kubectl: v0.23.5 => 0.24.0
 
+<aside class="note warning">
```
Comment (sbueringer) [inline docs/book/src/developer/providers/v1.1-to-v1.2.md:29]: Given that this doc is more like a provider migration guide. I'm not sure if we should add the note here. But no strong objections from my side, just feels a bit out of place

File: docs/book/src/user/troubleshooting.md [hunk 2]
```diff
@@ -116,6 +116,32 @@ Alternatively a Cert Manager yaml file can be placed in the [clusterctl override
...
+Upgrading Cert Manager may fail due to a breaking change introduced in Cert Manager release v1.6.
+An upgrade using `clusterctl` is affected when:
+
+* using `clusterctl` in version `v1.1.4` or a more recent version.
+* Cert Manager lower than version `v1.0.0` did run in the management cluster (which was shipped in Cluster API until including `v0.3.14`).
+
+This will cause `clusterctl upgrade apply` to fail with the error:
+
+```bash
+cluster-api % clusterctl upgrade apply
```
Comment (sbueringer) [inline docs/book/src/user/troubleshooting.md:130]:
```suggestion
clusterctl upgrade apply
```
nit. I think (xref: https://github.com/kubernetes-sigs/cluster-api/pull/6521/files)

File: docs/book/src/user/troubleshooting.md [hunk 3]
```diff
@@ -116,6 +116,32 @@ Alternatively a Cert Manager yaml file can be placed in the [clusterctl override
...
+```
+
+```bash
+clusterctl upgrade apply --contract v1beta1
```
Comment (sbueringer) [inline docs/book/src/user/troubleshooting.md:134]: duplicate to above? Do we need --contract above?

File: docs/book/src/user/troubleshooting.md (hunk 2 above)
Comment (chrischdi) [reply docs/book/src/user/troubleshooting.md:130]: Its the same as on that PR (or at the paragraph above)

File: docs/book/src/developer/providers/v1.1-to-v1.2.md (hunk 1 above)
Comment (chrischdi) [reply docs/book/src/developer/providers/v1.1-to-v1.2.md:29]: no bigger opinions on that from my side.

Comment (chrischdi) [reply docs/book/src/developer/providers/v1.1-to-v1.2.md:29]: no bigger opinions on that from my side. For me it is the link between the affected version to the troubleshooting section.

Comment (chrischdi) [reply docs/book/src/developer/providers/v1.1-to-v1.2.md:29]: Without any kind of such linkage a user would just fail and then hopefully find the troubleshooting section.

This way users which read the migration guide should get aware of the issue.

Comment (sbueringer) [reply docs/book/src/developer/providers/v1.1-to-v1.2.md:29]: Yup it's just that this document is the documentation for infra provider authors on how to migrate their infra provider.

So I wouldn't expect end users to find warnings that way. I guess this mostly shows a lack of release notes apart from the auto-generated ones where we can add "custom notes" over time :)

File: docs/book/src/user/troubleshooting.md (hunk 2 above)
Comment (sbueringer) [reply docs/book/src/user/troubleshooting.md:130]: I think the prefix is actually only above in this doc, right? As far as I can see the `cluster-api %` is not used anywhere else

File: docs/book/src/developer/providers/v1.1-to-v1.2.md (hunk 1 above)
Comment (chrischdi) [reply docs/book/src/developer/providers/v1.1-to-v1.2.md:29]: 🤔 fair. So we should we remove the cross reference then I think?

**Comment by fabriziopandini** [comment]
/lgtm
/approve 
We are working to automate the migration steps, but it is great to have this documentation in the meantime

**Pull request merged**